action/selection/open_spiel/database.cc

```diff
@@ -255,6 +279,10 @@ struct ExplainMicroserviceCost {
   // to process the entire EXPLAIN query tree. Inference results are only gathered when
   // record_predictions is enabled.
   std::stringstream inference_results_;
+
+  // The JSON document that is used to construct the request to the microservice. This is used
+  // so we can request inference for the entire EXPLAIN tree in a single request.
```

Comment:
At risk of incurring the disapproval of the spirit of a bearded man for letting hardware details leak in here,
I do think it is useful to specify why we did this in the code comments.

```suggestion
  // so we can request inference for the entire EXPLAIN tree in a single request.
  // We observed 2x speedup on dev4 (1 hour -> 30 minutes) by doing so.
```